tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -296,6 +315,32 @@ export function benchmarkMemory(testObject: IMemoryTestObject): Test {
 						heapUsedStats.marginOfErrorPercent > args.maxRelativeMarginOfError
 					);
 
+					if (baselineMemoryUsage >= 0 && allowedDeviationBytes >= 0) {
+						// Compare the average heap used to the baseline memory usage
+						const avgHeapUsed = heapUsedStats.arithmeticMean;
+						const lowerBound = baselineMemoryUsage - args.allowedDeviationBytes;
+						const upperBound = baselineMemoryUsage + args.allowedDeviationBytes;
+						const tolerance = (args.allowedDeviationBytes / baselineMemoryUsage) * 100;
+						// Throw errors on regressions/improvements if `ENABLE_MEM_REGRESSION` is set and a warning otherwise.
+						// This allows us to run the tests in CI without failing the build, but still get a warning.
+						if (avgHeapUsed > upperBound) {
+							const message = `Memory Regression detected for ${testObject.title}: Used ${avgHeapUsed} bytes, exceeding the ${tolerance}% tolerance of the baseline of ${args.baselineMemoryUsage} bytes.`;
```

Comment:
Related to previous comment. And adding quotes around all interpolated values so the message is easily readable if one of the values ends up being blank.
```suggestion
							const message = `Memory Regression detected for test '${testObject.title}'. Used '${avgHeapUsed}' bytes, with baseline '${args.baselineMemoryUsage}' and tolerance of '${allowedDeviationBytes}' bytes.`;
```